### DISCUSSION TITLE
[Question] How does the unmount feature work?

### DISCUSSION THREAD

**Discussion by Unb0rn** [tiann/KernelSU#1126] [Q&A] [answered]
I've finally set KernelSU up so I wanted to use a couple of modules and that raised a couple of questions.
First of all, I tested the system with Momo - without any modules KernelSU is invisible even if there are some superusers in the system. This works as expected and it's cool!
Then I wanted to play with some modules I used to add with Magisk - added the Wireguard for Android module (from the official app) that basically binds wg and wg-quick in the system. And that was immediately detected by Momo.
Even more, if I try to do something like `cat /proc/self/mounts` from non-ksu-enabled term I can see the KSU mount here.
Adding ZygiskNext makes it even worse - additional zygisk mount appears. Is it an expected behavior? Or maybe something is wrong with my KernelSU installation.

Of course, I have "Unmount modules by default" switched on (but I think it is on by default) and trying to add unmount per-app doesn't change this behavior too.

Installing Shamiko along with Zygisk triggers the unmounts but shouldn't we have this behavior with global unmount by default?

Thanks!

**Comment by tiann** [comment, marked as answer]
The global umount only works on GKI kernel, for non-GKI kernel, you need modules to achieve that.

[a comment by an automated account is not shown]

**Reply by Unb0rn** [reply]
My core_hook.c is pretty different, are we based on the same version? I can see ksu_can_umount and ksu_path_umount for kernels version 4.4 and higher for example

**Reply by Unb0rn** [reply]
Added the whole diff from wxt1221/KernelSU:main kernel/core_hook.c - umount still doesn't work for me. But the idea is great!)
